### PR TITLE
Create an async experiment for the linkables endpoint.

### PR DESCRIPTION
**Pull request by elliotcm**
We were going to use Github Scientist for this but
they avoid the async complexity by running everything
sequentially.  This means that our linkables
endpoint would take twice as long to respond while
we were running the test.

Scientist doesn't actually do much (you still have
to write your comparators and reporting) so I
wrote an async version.

Here's the flow:
fullsizerender

I'd love to get feedback from (at least):
- [x] @danielroseman 
- [ ] @tijmenb 
- [x] @dougdroper 
- [ ] @steventux

### CODE REVIEW

**Comment by elliotcm** [comment]
After some feedback from @dougdroper I've restructured the way the workers report the data to avoid locking issues.

Now only the control branch will try to report and will retry if the candidate isn't ready.

File: app/candidates/base_candidate.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    def experiment_candidate(experiment)
+      start_time = Time.now
+      result = yield
+      duration = (Time.now - start_time).to_f
+
+      ExperimentResultWorker.perform_async(experiment[:name], experiment[:id], result, duration, :candidate)
+
+      result
+    rescue StandardError => exception
+      if Rails.env.test?
```
Comment (boffbowsh) [inline app/candidates/base_candidate.rb]: Maybe dev too?

**Comment by boffbowsh** [comment]
This is good, I like the API. I think this will be really useful :) 👍

**Comment by dougdroper** [comment]
I agree 👍

File: app/candidates/base_candidate.rb (hunk 1 above)
Comment (elliotcm) [inline app/candidates/base_candidate.rb]: I'll make it configurable in puppet, defaulting to re-raise.

Comment (elliotcm) [inline app/candidates/base_candidate.rb]: Or maybe the other way round.

**Comment by elliotcm** [comment]
@boffbowsh I've changed it to an environment variable and enable it for test.  People who need it can activate it or if we find we need it all the time we can add it to the dev VM in puppet.

**Comment by elliotcm** [comment]
We've had enough top qual feedback I think, I'll get this merged and we can tweak it as we go if needed.

**Pull request merged**